9c-internal/chart/templates/service.yaml

```diff
@@ -44,13 +44,39 @@ spec:
   - port: {{ $.Values.miner.ports.graphql }}
     targetPort: {{ $.Values.miner.ports.graphql }}
     name: gql
-  {{- if $.Values.pos.enabled }}
-  - port: {{ $.Values.pos.port }}
-    targetPort: {{ $.Values.pos.port }}
+  selector:
+    app: miner-{{ $index }}
+  type: LoadBalancer
+
+---
+{{ end }}
+
+{{ range $idx := until (int .Values.validator.count) }}
+{{ $index := add $idx 1 }}
+apiVersion: v1
+kind: Service
+metadata:
+  name: validator-{{ $index }}
+  namespace: {{ $.Chart.Name }}
+  annotations:
+    service.beta.kubernetes.io/aws-load-balancer-scheme: internet-facing
+    service.beta.kubernetes.io/aws-load-balancer-type: "external"
+    service.beta.kubernetes.io/aws-load-balancer-nlb-target-type: "ip"
+spec:
+  ports:
+  - port: {{ $.Values.validator.ports.headless }}
+    targetPort: {{ $.Values.validator.ports.headless }}
+    name: headless
+  - port: {{ $.Values.validator.ports.graphql }}
+    targetPort: {{ $.Values.validator.ports.graphql }}
+    name: gql
+  {{- if eq $.Values.consensusType "pbft" }}
+  - port: {{ $.Values.validator.port }}
+    targetPort: {{ $.Values.validator.port }}
```

Comment:
i think it should be like `{{ $.Values.validator.ports.gossip }}`, depending on `values.yaml`

Reply:
amended!